### DISCUSSION TITLE
How to open an article without navigation?

### DISCUSSION THREAD

**Discussion by softshape** [facebook/docusaurus#5676] [Q&A]
I need to open a single article in popup, without header, sidebar, and footer. Is there any way to show an article with navigation hidden?

**Comment by slorber** [comment]
Docusaurus generates static files and does not have a "server" which means that we have 2 choices:
- emit 2 HTML files for each doc, one for regular usage, and one for "lightweight" usage
- emit 1 HTML file but find a way to detect it is embedded.

Emitting 2 files is likely to increase build times significantly (unless you can choose specifically for which docs 2 files must be emitted, with a frontmatter flag?), it complicates things a little bit (need to be built for each content plugin?) and the embedded form we render may be too opinionated to all use-cases.

What I suggest is that you inject some JS script into your doc, so that a `html.in-iframe` class is added: https://stackoverflow.com/questions/4353933/show-page-differently-css-if-within-iframe/25572943

Then you can hide the elements that you want to remove from this embedded view with plain CSS selectors/rules like `html.in-iframe`, for example rendering the content with `static` positioning over the rest.

This may be a bit hacky but can probably solve your use-case for now

**Reply by slorber** [reply]
Note: we already have a `print` mode so that docs can be printed without sidebar, TOC etc... 

It's not so great currently (particularly in dark mode) but maybe we can improve it and make it easier to turn this mode on so that we can embed any doc with a single querystring?

![image](https://user-images.githubusercontent.com/749374/143066766-7d86e7ad-7712-4c72-893d-321c6d775729.png)

**Comment by slorber** [comment]
Backlinking related issue: https://github.com/facebook/docusaurus/issues/7480
